src/Lexer.php

```diff
@@ -428,12 +428,7 @@ private function stripcslashes(string $str, string $quoteType): string
 
             if (isset(self::SPECIAL_CHARS[$nextChar])) {
                 $result .= self::SPECIAL_CHARS[$nextChar];
-            } elseif ($nextChar === '\\') {
-                $result .= $nextChar;
-            } elseif ($nextChar === "'" || $nextChar === '"') {
-                if ($nextChar !== $quoteType) {
-                    trigger_deprecation('twig/twig', '3.12', 'Character "%s" at position %d does not need to be escaped anymore.', $nextChar, $i + 1);
```

Comment:
I think I'm confused because of the deprecation message.

Maybe it's just me, but I read the current message like this: "The character does not need to be escaped, **but that's fine**.". And of course, it's not fine as the behavior will change, so the developer must fix the code.

So, maybe we need to add some hints to the developer:

Character "%s" at position %d should not to be escaped; the "\\" character is ignored in Twig 3 but as it won't be in 4.x, please remove the extra "\\" character.

Reply:
You're right. The deprecation is vague. 

Fixed in https://github.com/twigphp/Twig/pull/4219